CMakeLists.txt

```diff
@@ -13,9 +14,47 @@
 #
 
 cmake_minimum_required(VERSION 3.5)
-project(BoostConfig LANGUAGES CXX)
+project(BoostConfig VERSION 1.69 LANGUAGES CXX)
```

Comment:
I can pretty much guarantee that this will become incorrect if not tested. In fact it already is.

Reply:
See https://github.com/boostorg/assert/blob/develop/test/CMakeLists.txt#L6 and https://github.com/boostorg/assert/blob/develop/test/check_cmake_version.cpp for one possible approach.

Reply:
Longer term, if the components are decoupled and usable independently, the versioning can then diverge between components.

I'm not sure how the links you provided make a solution though.  Do you want to build and run a test program to get the version?  If so, would this not prove problematic for cross compilation?

Would it not be possible to copy the logic here: https://github.com/Kitware/CMake/blob/master/Modules/FindBoost.cmake#L1370 so that we always use the version parsed directly from the headers?